app/service/rest.py

```diff
@@ -63,11 +64,14 @@ def get_services():
     detailed = request.args.get('detailed') == 'True'
     user_id = request.args.get('user_id', None)
     include_from_test_key = request.args.get('include_from_test_key', 'True') != 'False'
+    start_date = request.args.get('start_date', None)
+    end_date = request.args.get('end_date', None)
```

Comment:
do we ever expect these to be `None`?

i'm not sure what comparing a date to null does in sql, but i don't like the sound of it.

in the `elif detailed:` block i'd put a  `if not start_date or not end_date: abort(400)` or whatever (i'll leave error message wording as an exercise to the reader)